services/platform/daemon/host/settings.go

```diff
@@ -2,88 +2,133 @@ package host
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"io"
+	"os"
 	"os/exec"
-	"strings"
 
 	v1 "github.com/home-cloud-io/core/api/platform/daemon/v1"
 	"github.com/home-cloud-io/core/services/platform/daemon/execute"
 
 	"github.com/steady-bytes/draft/pkg/chassis"
 )
 
+type (
+	TimeConfig struct {
+		TimeZone string `json:"timeZone"`
+	}
+	ServicesConfig struct {
+		Resolved struct {
+			Enable  bool     `json:"enable"`
+			Domains []string `json:"domains"`
+		} `json:"resolved"`
+		K3s struct {
+			Enable     bool   `json:"enable"`
+			Role       string `json:"role"`
+			ExtraFlags string `json:"extraFlags"`
+		} `json:"k3s"`
+		OpenSSH struct {
+			Enable         bool `json:"enable"`
+			AuthorizedKeys struct {
+				Keys []string `json:"keys"`
+			} `json:"authorizedKeys"`
+		} `json:"openssh"`
+		Avahi struct {
```

Comment:
Hmm what is this structure?

Reply:
I'll add a comment explaining them. It's a part of this shift in the OS to use `.json` based configuration instead of `.nix` configuration files.

https://github.com/home-cloud-io/os/pull/2